### PR TITLE
SS-146 Public apps view

### PR DESCRIPTION
**Pull request by alfredeen**
## Status

- [x] Ready
- [ ] Draft
- [ ] Hold

## Description

Reworked the public apps view. Added back the description field to the AppInstance model which can now be used by the user on app create, app update and displayed on the app cards. Also styled the app theme tags.

Jira [link](https://scilifelab.atlassian.net/browse/SS-146)

## Types of changes

What types of changes does your code introduce to STACKn?

- [ ] Hotfix (fixing a critical bug in production)
- [ ] Bugfix
- [x] New feature
- [ ] Documentation update

## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have included migration files (if there are changes to the model classes)
- [ ] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [ ] I have updated the release notes (releasenotes.md)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

File: templates/portal/apps.html [hunk 1]
```diff
@@ -7,87 +7,119 @@
...
+                    <div class="card h-100 border-0 shadow" style="min-width: 300px;">
+                        <div class="card-body d-flex flex-column mx-4 my-1">
+                            <div class="col-12 d-flex flex-row pb-2 align-items-center">
+                                <div class="col-8 d-flex">
+                                    <h5 class="card-title bg-white">{{ app.name }}</h5>
+                                </div>
+                                <div class="col-4 d-flex justify-content-end">
+                                    {% static 'images/logos/apps/' as static_url %}
+                                    <img class="img-fluid float-end img-cover" style="height:50px;padding-right: 5%;"
+                                        src="{{static_url}}{{app.app.slug}}-logo.png"
```
Comment (sandstromviktor) [inline templates/portal/apps.html:45]: I think this is not the same as used in the helm charts for installing the apps. For instance, the `Standard App` uses `default-logo.png`. This needs to be solved somehow. Perhaps it is possible to fetch the "logo" information from the installed app in the database?

File: templates/portal/apps.html [hunk 2]
```diff
@@ -7,87 +7,119 @@
...
+                <div class="col-12 col-sm-12 col-md-6 col-lg-6 col-xl-4 col-xxl-4 mb-4 pb-4">
+                    <div class="card h-100 border-0 shadow" style="min-width: 300px;">
+                        <div class="card-body d-flex flex-column mx-4 my-1">
+                            <div class="col-12 d-flex flex-row pb-2 align-items-center">
+                                <div class="col-8 d-flex">
+                                    <h5 class="card-title bg-white">{{ app.name }}</h5>
+                                </div>
+                                <div class="col-4 d-flex justify-content-end">
+                                    {% static 'images/logos/apps/' as static_url %}
+                                    <img class="img-fluid float-end img-cover" style="height:50px;padding-right: 5%;"
```
Comment (sandstromviktor) [inline templates/portal/apps.html:44]: If you accept the `m-1` suggestion from above, then i would suggest removing `padding-right` here.

File: templates/portal/apps.html [hunk 3]
```diff
@@ -7,87 +7,119 @@
...
+                                    {% if request.session.app_tags|exists:app.id %}
+                                    {% with app.tags|split:"," as tags %}
+                                    {% for tag in tags %}
+                                    <a class="tag {% if tag in request.session.app_tag_filters%}disabled{% endif %}"
+                                        href="{% url 'portal:apps' %}?tf_add={{ tag | urlencode }}">
+                                        <div class="tag-ico fas fa-tag fa-sm"></div><span class="badge text-bg-primary">{{ tag }}</span>
+                                    </a>
+                                    {% endfor %}
+                                    <a href="{% url 'portal:apps' %}?app_id_remove={{ app.id | urlencode }}&tag_count={{ tags.count | urlencode }}"
+                                        style="display: inline-flex;">
```
Comment (sandstromviktor) [inline templates/portal/apps.html:71]:
```suggestion
                                        class="d-inline-flex">
```

File: templates/portal/apps.html [hunk 4]
```diff
@@ -7,87 +7,119 @@
...
+                                    <a class="tag {% if tag in request.session.app_tag_filters%}disabled{% endif %}"
+                                        href="{% url 'portal:apps' %}?tf_add={{ tag | urlencode }}">
+                                        <div class="tag-ico fas fa-tag fa-sm"></div><span class="badge text-bg-primary">{{ tag }}</span>
+                                    </a>
+                                    {% endfor %}
+                                    {% if tags.count > tag_limit %}
+                                    <a href="{% url 'portal:apps' %}?app_id_add={{ app.id | urlencode }}&tag_count={{ tags.count | urlencode }}"
+                                        style="display: inline-flex;">
+                                        {% csrf_token %}
+                                        <button style="font-weight: 600;" type="submit" class="btn btn-light tag-count">+{{
```
Comment (sandstromviktor) [inline templates/portal/apps.html:91]:
```suggestion
                                        <button type="submit" class="btn btn-light fw-600 tag-count">+{{
```

File: templates/portal/apps.html [hunk 5]
```diff
@@ -7,87 +7,119 @@
...
+                                    {% with tags|count_str as tag_limit %}
+                                    {% for tag in tags|slice:tag_limit %}
+                                    <a class="tag {% if tag in request.session.app_tag_filters%}disabled{% endif %}"
+                                        href="{% url 'portal:apps' %}?tf_add={{ tag | urlencode }}">
+                                        <div class="tag-ico fas fa-tag fa-sm"></div><span class="badge text-bg-primary">{{ tag }}</span>
+                                    </a>
+                                    {% endfor %}
+                                    {% if tags.count > tag_limit %}
+                                    <a href="{% url 'portal:apps' %}?app_id_add={{ app.id | urlencode }}&tag_count={{ tags.count | urlencode }}"
+                                        style="display: inline-flex;">
```
Comment (sandstromviktor) [inline templates/portal/apps.html:89]:
```suggestion
                                        class="d-inline-flex">
```

**Review by sandstromviktor — CHANGES_REQUESTED** [review]
Great job! Some minor things to fix, if you agree. Don't use the comments as "correct answers".

File: templates/portal/apps.html [hunk 6]
```diff
@@ -7,87 +7,119 @@
...
-        <div class="card h-100 border-0 shadow">
 
-            <div style="height:350px;" class="align-items-center d-flex justify-content-center p-2">
+    <div class="row pt-3 g-4">
+        <div class="card-group">
+            <div class="row row-cols-1 row-cols-md-3 g-4">
+                {% for app in published_apps %}
+                <div class="col-12 col-sm-12 col-md-6 col-lg-6 col-xl-4 col-xxl-4 mb-4 pb-4">
+                    <div class="card h-100 border-0 shadow" style="min-width: 300px;">
+                        <div class="card-body d-flex flex-column mx-4 my-1">
```
[the remark this answers is not in the document]
Comment (alfredeen) [reply templates/portal/apps.html:37]: Yes. I like m-1. Coming in a combined commit.

**Review by sandstromviktor — APPROVED** [review]
Really good job!:)

**Comment by akochari** [comment]
Just looked through everything, I think it looks great. Maybe we can add a test to check that making an app public actually results in the app being displayed on the public apps page?

**Pull request merged**